Services/Awareness/classes/class.WidgetManager.php

```diff
@@ -79,37 +83,37 @@ public function notifyOnNewOnlineContacts() : void
         foreach ($d as $u) {
             $uname = "[" . $u->login . "]";
             if ($u->public_profile) {
-                $uname = "<a href='./goto.php?target=usr_" . $u->id . "'>" . $u->lastname . ", " . $u->firstname . " " . $uname . "</a>";
+                $uname = $u->lastname . ", " . $u->firstname . " " . $uname;
             }
             if (!in_array($u->id, $no_ids)) {
-                $new_online_users[] = $uname;
+                $new_online_users[] = new ilNotificationLink(
+                    new ilNotificationParameter($uname),
+                    './goto.php?target=usr_' . $u->id
+                );
                 $no_ids[] = $u->id;
             }
         }
         if (count($new_online_users) == 0) {
             return;
         }
-        //var_dump($d); exit;
-        $lng->loadLanguageModule('mail');
 
-        //$recipient = ilObjectFactory::getInstanceByObjId($this->user_id);
-        $bodyParams = array(
-            'online_user_names' => implode("<br />", $new_online_users)
-        );
-        $notification = new \ilNotificationConfig('osd_main');
-        $notification->setTitleVar('awareness_now_online', $bodyParams, 'awrn');
-        $notification->setShortDescriptionVar('awareness_now_online_users', $bodyParams, 'awrn');
-        $notification->setLongDescriptionVar('', $bodyParams, '');
-        $notification->setAutoDisable(false);
-        //$notification->setLink();
+        $notification = new \ilNotificationConfig('who_is_online');
+        $notification->setTitleVar('awareness_now_online', [], 'awrn');
+        $notification->setShortDescriptionVar('');
+        $notification->setLongDescriptionVar('');
+        $notification->setLinks($new_online_users);
         $notification->setIconPath('templates/default/images/icon_usr.svg');
         $notification->setValidForSeconds(\ilNotificationConfig::TTL_SHORT);
         $notification->setVisibleForSeconds(\ilNotificationConfig::DEFAULT_TTS);
 
-        //$notification->setHandlerParam('mail.sender', $sender_id);
-
         $this->session_repo->setOnlineUsersTS(date("Y-m-d H:i:s", time()));
-        
+
+        foreach (ilNotificationOSDHandler::getNotificationsForUser($this->user_id) as $osd) {
```

Comment:
Could we already use ave some instance methods instead?

Static methods are hard to tests and dependencies are gathered by the usage  of the `Global Scope`. I'd prefer some kind of [Repository](https://github.com/ILIAS-eLearning/ILIAS/blob/trunk/docs/development/repository-pattern.md). For me it is okay do refactor this in the beta phase.

The type should be passed to the service or repository. Filtering this in the `User Land` is not the best approach IMO.

Reply:
This will be part of a following major refactoring/PHP8 PR